### ISSUE TITLE
[FormRequest] recovering validated fields not working as expected (nested fields)

### ISSUE THREAD

**Issue by lk77** [laravel/framework#23612]
Hello,

currently, it seems that it's not possible to recover validated fields correctly where some fields are nested, example : 

->rules()
```
'data.field' => 'required'
```
->all()
```
'data' => [
    'field' => 'val', 
    'false_field' => 'val'
]
```
->validated()
```
'data' => [
    'field' => 'val', 
    'false_field' => 'val'
]
```
i was expecting to get :
```
'data' => [
    'field' => 'val'
]
```
the only way i found is to override the validated method but it's not a solution,
perhaps we could have a syntax like `data->field` to handle this, i don't know.

thanks.

**Comment by staudenmeir** [comment]
Nested rules get reduced to the first key for returning the validated data:
https://github.com/laravel/framework/blob/5.6/src/Illuminate/Foundation/Http/FormRequest.php#L173

This was added by #20974 to handle wildcard rules but affects all nested rules.

Unfortunately we can't just replace `Str::contains($rule, '.')` with `Str::contains($rule, '*')` because `Arr::only()` doesn't support nested keys.

**Comment by lk77** [comment]
Ok,
i've used a custom rule to handle this, it's less flexible because it throw the request away, but it works.
thanks.

**Comment by staudenmeir** [comment]
It will be fixed in 5.7: #23708

**Comment by lk77** [comment]
thanks !
